portfolio/src/main/java/com/google/sps/servlets/CommentsHandler.java

```diff
@@ -88,11 +88,34 @@ public String getComments() {
     return gson.toJson(commentsMap.values());
   }
 
-  public void deleteComments() {
+  // deletes the comment with the given key
+  // deletes all child comments
+  public void deleteComment(String key) {
+    // Technically if there are replies to this comment I can delete the whole comments thread, I don't know
+    // what is better - to delete them all or to change parent comment to dummy one with "deleted" text
+    // Here I am implementing the first one
```

Comment:
Yeah I agree. I have seen systems that do both. I always felt deleting entire thread is more intuitve but then the content is lost. And second way, we lose the context because parent comment is lost, but it preserves content.

Reply:
At this point it becomes a product decision and I'm happy with either way :)